src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/WorkflowRunRepository.kt

```diff
@@ -21,15 +28,45 @@ class OrderlyWebWorkflowRunRepository : WorkflowRunRepository
     {
         JooqContext().use {
             it.dsl.insertInto(Tables.ORDERLYWEB_WORKFLOW_RUN)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.NAME, workflowRun.name)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.KEY, workflowRun.key)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.EMAIL, workflowRun.email)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.DATE, Timestamp.from(workflowRun.date))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.REPORTS, Gson().toJson(workflowRun.reports))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.INSTANCES, Gson().toJson(workflowRun.instances))
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH, workflowRun.gitBranch)
-                .set(Tables.ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT, workflowRun.gitCommit)
-                .execute()
+                    .set(ORDERLYWEB_WORKFLOW_RUN.NAME, workflowRun.name)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.KEY, workflowRun.key)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.EMAIL, workflowRun.email)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.DATE, Timestamp.from(workflowRun.date))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.REPORTS, Gson().toJson(workflowRun.reports))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.INSTANCES, Gson().toJson(workflowRun.instances))
+                    .set(ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH, workflowRun.gitBranch)
+                    .set(ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT, workflowRun.gitCommit)
+                    .execute()
+        }
+    }
+
+    override fun getWorkflowRunDetails(key: String): WorkflowRun
+    {
+        JooqContext().use {
+            val result = it.dsl.select(
+                    ORDERLYWEB_WORKFLOW_RUN.NAME,
+                    ORDERLYWEB_WORKFLOW_RUN.KEY,
+                    ORDERLYWEB_WORKFLOW_RUN.EMAIL,
+                    ORDERLYWEB_WORKFLOW_RUN.DATE,
+                    ORDERLYWEB_WORKFLOW_RUN.REPORTS,
+                    ORDERLYWEB_WORKFLOW_RUN.INSTANCES,
+                    ORDERLYWEB_WORKFLOW_RUN.GIT_BRANCH,
+                    ORDERLYWEB_WORKFLOW_RUN.GIT_COMMIT)
+                    .from(ORDERLYWEB_WORKFLOW_RUN)
+                    .where(ORDERLYWEB_WORKFLOW_RUN.KEY.eq(key))
+                    .singleOrNull()
+                    ?: throw UnknownObjectError("key", "workflow")
```

Comment:
```suggestion
                    ?: throw UnknownObjectError(key, "workflow")
```

Reply:
I have left this key in this way as I do not want to display it in the error message.

Reply:
It should be included - it's important for debugging and we do the same thing everywhere else in the app

Reply:
We haven't shown key anywhere else and I think is it not a good idea to show api key especially from security point of view.

Reply:
I agree with Mark on this - we don't tend to show api keys to the user, but it's not really a security issue - we're passing them around freely in the urls after all. All other users of UnknownObjectError include the unknown value to aid debug and should do the same here really.

Reply:
Done